pkg/webui/components/key-value-map/entry.js

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+
+import PropTypes from '../../lib/prop-types'
+import Input from '../input'
+import Button from '../button'
+
+@bind
+class Entry extends React.Component {
+
+  handleRemoveButtonClicked (event) {
+    this.props.onRemoveButtonClick(this.props.index, event)
+  }
+
+  handleKeyChanged (newKey) {
+    this.props.onChange(this.props.index, { key: newKey })
+  }
+
+  handleValueChanged (newValue) {
+    this.props.onChange(this.props.index, { value: newValue })
+  }
+
+  render () {
+    return (
+      <Row key={this.props.index}>
+        <Col sm={4}>
+          <Input
+            name={`${this.props.name}[${this.props.index}].key`}
+            placeholder={this.props.keyPlaceholder}
+            type="text"
+            onChange={this.handleKeyChanged}
+            value={this.props.value.key}
+            required
+          />
+        </Col>
+        <Col sm={7}>
+          <Input
+            name={`${this.props.name}[${this.props.index}].value`}
+            placeholder={this.props.valuePlaceholder}
+            type="text"
+            onChange={this.handleValueChanged}
+            value={this.props.value.value}
+            required
+          />
+        </Col>
+        <Col sm={1}>
+          <Button
+            name={`${this.props.name}[${this.props.index}].remove`}
+            type="button"
+            onClick={this.handleRemoveButtonClicked}
+            icon="delete"
```

Comment:
```suggestion
            icon="delete"
            danger
```
Will change the button color to red.